bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,198 @@
+        """Generate a 2d array for the board."""
+        board: typing.List[typing.List[typing.Union[str, int]]] = [
+            ["bomb" if random() <= bomb_chance else "number" for _ in range(10)] for _ in range(10)]
+        for y, row in enumerate(board):
+            for x, cell in enumerate(row):
+                if cell == "number":
+                    # calculate bombs near it
+                    to_check = []
+                    for xt in [x - 1, x, x + 1]:
+                        for yt in [y - 1, y, y + 1]:
+                            if xt != -1 and xt != 10 and yt != -1 and yt != 10:
+                                to_check.append(board[yt][xt])
+
+                    bombs = sum(map(self.is_bomb, to_check))
+                    board[y][x] = bombs
+        return board
+
+    @staticmethod
+    def format_for_discord(board: typing.List[typing.List[typing.Union[str, int]]]) -> str:
+        """Format the board to a string for discord."""
+        mapping = {
+            0: ":stop_button:",
+            1: ":one:",
+            2: ":two:",
+            3: ":three:",
+            4: ":four:",
+            5: ":five:",
+            6: ":six:",
+            7: ":seven:",
+            8: ":eight:",
+            9: ":nine:",
+            10: ":keycap_ten:",
+            "bomb": ":bomb:",
+            "hidden": ":grey_question:",
+            "flag": ":triangular_flag_on_post:"
+        }
+
+        discord_msg = ":stop_button:    :regional_indicator_a::regional_indicator_b::regional_indicator_c:" \
+                      ":regional_indicator_d::regional_indicator_e::regional_indicator_f::regional_indicator_g:" \
+                      ":regional_indicator_h::regional_indicator_i::regional_indicator_j:\n\n"
+        rows: typing.List[str] = []
+        for row_number, row in enumerate(board):
+            new_row = mapping[row_number + 1] + "    "
+            for cell in row:
+                new_row += mapping[cell]
+            rows.append(new_row)
+
+        discord_msg += "\n".join(rows)
+        return discord_msg
+
+    @commands.command(name="minesweeper")
+    async def minesweeper_command(self, ctx: commands.Context, bomb_chance: float = .2) -> None:
+        """Start a game of minesweeper."""
+        if ctx.author in self.games.keys():  # Player is already playing
+            msg = await ctx.send(f"{ctx.author.mention} you already have a game running")
+            await msg.delete(delay=2)
+            await ctx.message.delete(delay=2)
+            return
+
+        # Add game to list
+        board = self.generate_board(bomb_chance)
+        reveled_board = [["hidden" for _ in range(10)] for _ in range(10)]
+
+        await ctx.send(f"{ctx.author.mention} is playing minesweeper")
+        chat_msg = await ctx.send(self.format_for_discord(reveled_board))
+
+        await ctx.author.send("play by typing: `.reveal xy xy ...` or `.flag xy xy ...` \nclose the game with `.end`")
```

Comment:
This could probably be displayed ad `.reveal xy [xy]` or something similar to display it's optional or just say you can use as many coordinates as you want with the command